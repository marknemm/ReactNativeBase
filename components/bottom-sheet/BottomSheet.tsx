import Backdrop from '@components/backdrop/Backdrop';
import { CLOSE_ICON } from '@constants/icons';
import { useModal } from '@hooks/modal-hooks';
import { useCallbacks } from '@hooks/state-hooks';
import { Button, BottomSheet as RneBottomSheet, Text } from '@rneui/themed';
import { useCallback } from 'react';
import { View } from 'react-native';
import type { BottomSheetProps } from './BottomSheet.interfaces';
import { useStyles } from './BottomSheet.styles';

/**
 * A component for displaying a bottom sheet.
 *
 * @param props The {@link BottomSheetProps}.
 * @returns The {@link BottomSheet} component.
 */
const BottomSheet: React.FC<BottomSheetProps> = (props) => {
  const { backdropStyle, children, isVisible, onBackdropPress, onClose, title } = props;
  const styles = useStyles(props);

  useModal(isVisible, useCallback(() => // Show custom Backdrop to change animation from sliding up to fading in.
    <Backdrop style={backdropStyle} />,
  [backdropStyle]));

  return (
    <RneBottomSheet
      {...props}
      backdropStyle={styles.backdrop}
      containerStyle={styles.container}
      onBackdropPress={useCallbacks(onBackdropPress, onClose)}
    >
      <View style={styles.inner}>
        <View style={styles.header}>
          <Text style={styles.title}>
            { title }
          </Text>
          <View style={styles.closeButtonContainer}>
            <Button
              accessibilityLabel="Close"
              icon={CLOSE_ICON}
              onPress={onClose}
              size="sm"
              type="clear"
            />
          </View>
        </View>

        {children}
      </View>
    </RneBottomSheet>
  );
};

export type * from './BottomSheet.interfaces';
export default BottomSheet;
