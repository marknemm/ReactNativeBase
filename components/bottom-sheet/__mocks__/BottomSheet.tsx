import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
