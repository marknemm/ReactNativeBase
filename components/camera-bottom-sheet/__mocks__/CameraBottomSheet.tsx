import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
