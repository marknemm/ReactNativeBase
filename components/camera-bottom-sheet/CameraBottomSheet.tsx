import BottomSheet from '@components/bottom-sheet/BottomSheet';
import { CAMERA_ICON, PHOTOS_ICON } from '@constants/icons';
import { Icon, ListItem } from '@rneui/themed';
import delay from 'delay';
import type { CameraBottomSheetProps } from './CameraBottomSheet.interfaces';

/**
 * A {@link BottomSheet} for choosing a photo from the camera or media library.
 *
 * @param props The {@link CameraBottomSheetProps}.
 * @returns The {@link CameraBottomSheet} component.
 */
const CameraBottomSheet: React.FC<CameraBottomSheetProps> = ({
  isVisible,
  onClose,
  onPressChoosePhoto,
  onPressTakePhoto,
  title,
}) => (
  <BottomSheet
    isVisible={isVisible}
    onClose={onClose}
    title={title}
  >
    <ListItem
      onPress={async () => {
        onClose?.();
        await delay(200); // Prevent bottom sheet close from blocking camera opening
        onPressTakePhoto?.();
      }}
      role="button"
      bottomDivider
    >
      <ListItem.Content>
        <ListItem.Title>Take Photo</ListItem.Title>
      </ListItem.Content>
      <Icon {...CAMERA_ICON} color="grey" />
    </ListItem>
    <ListItem
      onPress={async () => {
        onClose?.();
        await delay(200); // Prevent bottom sheet close from blocking media library opening
        onPressChoosePhoto?.();
      }}
      role="button"
      bottomDivider
    >
      <ListItem.Content>
        <ListItem.Title>Choose Photo</ListItem.Title>
      </ListItem.Content>
      <Icon {...PHOTOS_ICON} color="grey" />
    </ListItem>
  </BottomSheet>
);

export type * from './CameraBottomSheet.interfaces';
export default CameraBottomSheet;
