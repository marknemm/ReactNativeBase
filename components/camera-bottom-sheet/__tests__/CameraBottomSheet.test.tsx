import CameraBottomSheet from '@components/camera-bottom-sheet/CameraBottomSheet';
import AppProvider from '@test/contexts/app/AppProvider';
import { act, render, screen, userEvent } from '@testing-library/react-native';
import BottomSheet from '@components/bottom-sheet/BottomSheet';

jest.mock('@components/bottom-sheet/BottomSheet');

describe('<CameraBottomSheet />', () => {
  describe('callbacks', () => {
    it('Calls onClose when the close button is pressed', () => {
      const onClose = jest.fn();
      render(
        <CameraBottomSheet
          isVisible
          onClose={onClose}
        />,
        { wrapper: AppProvider }
      );

      act(() =>
        (BottomSheet as jest.Mock).mock.lastCall[0]?.onClose()
      );

      expect(onClose).toHaveBeenCalled();
    });

    it('Calls onPressTakePhoto when the take photo button is pressed', async () => {
      const onClose = jest.fn();
      const onPressTakePhoto = jest.fn();
      render(
        <CameraBottomSheet
          isVisible
          onClose={onClose}
          onPressTakePhoto={onPressTakePhoto}
        />,
        { wrapper: AppProvider }
      );

      const takePhotoButton = screen.getByRole('button', { name: 'Take Photo' });
      await userEvent.press(takePhotoButton);
      await jest.runAllTimersAsync();

      expect(onClose).toHaveBeenCalled();
      expect(onPressTakePhoto).toHaveBeenCalled();
    });

    it('Calls onPressChoosePhoto when the choose photo button is pressed', async () => {
      const onClose = jest.fn();
      const onPressChoosePhoto = jest.fn();
      render(
        <CameraBottomSheet
          isVisible
          onClose={onClose}
          onPressChoosePhoto={onPressChoosePhoto}
        />,
        { wrapper: AppProvider }
      );

      const takePhotoButton = screen.getByRole('button', { name: 'Choose Photo' });
      await userEvent.press(takePhotoButton);
      await jest.runAllTimersAsync();

      expect(onClose).toHaveBeenCalled();
      expect(onPressChoosePhoto).toHaveBeenCalled();
    });
  });

  describe('snapshots', () => {
    it('renders correctly when not visible', () => {
      render(
        <CameraBottomSheet />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when visible', () => {
      render(
        <CameraBottomSheet isVisible />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
