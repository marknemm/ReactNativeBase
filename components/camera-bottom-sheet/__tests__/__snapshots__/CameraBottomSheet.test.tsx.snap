// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CameraBottomSheet /> snapshots renders correctly when not visible 1`] = `null`;

exports[`<CameraBottomSheet /> snapshots renders correctly when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "backgroundColor": "transparent",
        },
      ]
    }
    testID="RNE__Overlay__backdrop"
  />
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    pointerEvents="box-none"
    replaceTheme={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "column-reverse",
      }
    }
    theme={
      {
        "colors": {
          "background": "#ffffff",
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff3b30",
          "grey": "#7d7d7d",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "placeholder": "#86939e",
          "platform": {
            "android": {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#007aff",
          "searchBg": "#dcdce1",
          "secondary": "#5856d6",
          "success": "#4cd964",
          "warning": "#ffcc00",
          "white": "#ffffff",
        },
        "font": {
          "size": {
            "large": 28.285714285714285,
            "larger": 31.428571428571427,
            "normal": 23.57142857142857,
            "small": 18.857142857142858,
          },
          "weight": {
            "bold": "bold",
            "bolder": "700",
            "boldest": "900",
            "light": "300",
            "lighter": "200",
            "lightest": "100",
            "normal": "normal",
          },
        },
        "mode": "light",
        "spacing": {
          "lg": 18.857142857142858,
          "md": 12.571428571428571,
          "screenHorizontal": 18.857142857142858,
          "screenVertical": 37.714285714285715,
          "sm": 6.285714285714286,
          "xl": 37.714285714285715,
          "xs": 3.142857142857143,
        },
      }
    }
    updateTheme={[Function]}
  >
    <View>
      <RCTScrollView>
        <View>
          <View
            style={
              {
                "backgroundColor": "#ffffff",
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "opacity": 1,
                "paddingBottom": 56.57142857142857,
                "paddingTop": 18.857142857142858,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "height": 40,
                  "paddingHorizontal": 6.285714285714286,
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <Text
                accessibilityRole="text"
                style={
                  {
                    "color": "#242424",
                    "flexGrow": 1,
                    "fontFamily": undefined,
                    "fontSize": 28.285714285714285,
                    "fontStyle": "normal",
                    "fontWeight": "bold",
                    "marginBottom": 0,
                    "textAlign": "center",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                  }
                }
              />
              <View
                style={
                  {
                    "position": "absolute",
                    "right": 3.142857142857143,
                    "top": -3.142857142857143,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 3.142857142857143,
                      },
                      undefined,
                      false,
                    ]
                  }
                  testID="RNE_BUTTON_WRAPPER"
                >
                  <View
                    accessibilityLabel="Close"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": false,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "borderColor": "#007aff",
                          "borderRadius": 3.142857142857143,
                          "borderWidth": 0,
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 6.285714285714286,
                          "paddingHorizontal": 8.285714285714285,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "marginHorizontal": 5,
                            "overflow": "hidden",
                          }
                        }
                        testID="RNE__ICON__CONTAINER"
                      >
                        <View
                          accessibilityRole="button"
                          android_ripple={
                            {
                              "borderless": false,
                              "color": "rgba(0, 0, 0, 0.3)",
                              "radius": -5,
                            }
                          }
                          disabled={false}
                          testID="RNE__ICON__CONTAINER_ACTION"
                        >
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "backgroundColor": "transparent",
                                "justifyContent": "center",
                              }
                            }
                            testID="RNE__ICON"
                          >
                            <Text
                              allowFontScaling={false}
                              selectable={false}
                              style={
                                [
                                  {
                                    "color": "#242424",
                                    "fontSize": 24,
                                  },
                                  {
                                    "backgroundColor": "transparent",
                                  },
                                  {
                                    "fontFamily": "material",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  {},
                                ]
                              }
                              testID="RNE__ICON__Component"
                            >
                              
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              replaceTheme={[Function]}
              role="button"
              updateTheme={[Function]}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#ffffff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#bcbbc1",
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
                testID="RNE__LISTITEM__padView"
              >
                <View
                  replaceTheme={[Function]}
                  style={
                    [
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  theme={
                    {
                      "colors": {
                        "background": "#ffffff",
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "placeholder": "#86939e",
                        "platform": {
                          "android": {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                        "white": "#ffffff",
                      },
                      "font": {
                        "size": {
                          "large": 28.285714285714285,
                          "larger": 31.428571428571427,
                          "normal": 23.57142857142857,
                          "small": 18.857142857142858,
                        },
                        "weight": {
                          "bold": "bold",
                          "bolder": "700",
                          "boldest": "900",
                          "light": "300",
                          "lighter": "200",
                          "lightest": "100",
                          "normal": "normal",
                        },
                      },
                      "mode": "light",
                      "spacing": {
                        "lg": 18.857142857142858,
                        "md": 12.571428571428571,
                        "screenHorizontal": 18.857142857142858,
                        "screenVertical": 37.714285714285715,
                        "sm": 6.285714285714286,
                        "xl": 37.714285714285715,
                        "xs": 3.142857142857143,
                      },
                    }
                  }
                  updateTheme={[Function]}
                >
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "backgroundColor": "transparent",
                        "color": "#242424",
                        "fontSize": 17,
                      }
                    }
                    testID="listItemTitle"
                  >
                    Take Photo
                  </Text>
                </View>
                <View
                  style={
                    {
                      "paddingLeft": 16,
                    }
                  }
                />
                <View
                  style={
                    {
                      "overflow": "hidden",
                    }
                  }
                  testID="RNE__ICON__CONTAINER"
                >
                  <View
                    accessibilityRole="button"
                    android_ripple={
                      {
                        "borderless": false,
                        "color": "rgba(0, 0, 0, 0.3)",
                        "radius": -5,
                      }
                    }
                    disabled={false}
                    replaceTheme={[Function]}
                    testID="RNE__ICON__CONTAINER_ACTION"
                    updateTheme={[Function]}
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      testID="RNE__ICON"
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "grey",
                              "fontSize": 24,
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                            {
                              "fontFamily": "material",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                        testID="RNE__ICON__Component"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              replaceTheme={[Function]}
              role="button"
              updateTheme={[Function]}
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#ffffff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#bcbbc1",
                    "flexDirection": "row",
                    "padding": 14,
                  }
                }
                testID="RNE__LISTITEM__padView"
              >
                <View
                  replaceTheme={[Function]}
                  style={
                    [
                      {
                        "alignItems": "flex-start",
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                  theme={
                    {
                      "colors": {
                        "background": "#ffffff",
                        "black": "#242424",
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "placeholder": "#86939e",
                        "platform": {
                          "android": {
                            "error": "#f44336",
                            "grey": "rgba(0, 0, 0, 0.54)",
                            "primary": "#2196f3",
                            "searchBg": "#dcdce1",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "default": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "ios": {
                            "error": "#ff3b30",
                            "grey": "#7d7d7d",
                            "primary": "#007aff",
                            "searchBg": "#dcdce1",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                          "web": {
                            "error": "#ff190c",
                            "grey": "#393e42",
                            "primary": "#2089dc",
                            "searchBg": "#303337",
                            "secondary": "#ca71eb",
                            "success": "#52c41a",
                            "warning": "#faad14",
                          },
                        },
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                        "white": "#ffffff",
                      },
                      "font": {
                        "size": {
                          "large": 28.285714285714285,
                          "larger": 31.428571428571427,
                          "normal": 23.57142857142857,
                          "small": 18.857142857142858,
                        },
                        "weight": {
                          "bold": "bold",
                          "bolder": "700",
                          "boldest": "900",
                          "light": "300",
                          "lighter": "200",
                          "lightest": "100",
                          "normal": "normal",
                        },
                      },
                      "mode": "light",
                      "spacing": {
                        "lg": 18.857142857142858,
                        "md": 12.571428571428571,
                        "screenHorizontal": 18.857142857142858,
                        "screenVertical": 37.714285714285715,
                        "sm": 6.285714285714286,
                        "xl": 37.714285714285715,
                        "xs": 3.142857142857143,
                      },
                    }
                  }
                  updateTheme={[Function]}
                >
                  <Text
                    accessibilityRole="text"
                    style={
                      {
                        "backgroundColor": "transparent",
                        "color": "#242424",
                        "fontSize": 17,
                      }
                    }
                    testID="listItemTitle"
                  >
                    Choose Photo
                  </Text>
                </View>
                <View
                  style={
                    {
                      "paddingLeft": 16,
                    }
                  }
                />
                <View
                  style={
                    {
                      "overflow": "hidden",
                    }
                  }
                  testID="RNE__ICON__CONTAINER"
                >
                  <View
                    accessibilityRole="button"
                    android_ripple={
                      {
                        "borderless": false,
                        "color": "rgba(0, 0, 0, 0.3)",
                        "radius": -5,
                      }
                    }
                    disabled={false}
                    replaceTheme={[Function]}
                    testID="RNE__ICON__CONTAINER_ACTION"
                    updateTheme={[Function]}
                  >
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "transparent",
                          "justifyContent": "center",
                        }
                      }
                      testID="RNE__ICON"
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "grey",
                              "fontSize": 24,
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                            {
                              "fontFamily": "material",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            {},
                          ]
                        }
                        testID="RNE__ICON__Component"
                      >
                        
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
